python/sparkdl/image/imageIO.py

```diff
@@ -118,89 +82,75 @@ def imageStructToArray(imageRow):
-    if num_channels == 1:
-        if arr.dtype not in [np.float16, np.float32, np.float64]:
-            raise ValueError("incompatible dtype (%s) for numpy array for float32 mode" %
-                             arr.dtype.string)
-        return SparkMode.FLOAT32
-    elif num_channels != 3:
-        raise ValueError("number of channels of the input array (%d) is not supported" %
-                         num_channels)
-    elif arr.dtype == np.uint8:
-        return SparkMode.RGB
-    elif arr.dtype in [np.float16, np.float32, np.float64]:
-        return SparkMode.RGB_FLOAT32
+    :param imageRow: Row, must have ImageSchema
+    :return PIL image
+    """
+    imgType = imageTypeByOrdinal(imageRow.mode)
+    if imgType.dtype != 'uint8':
+        raise ValueError("Can not convert image of type " + imgType.dtype + " to PIL, can only deal with 8U format")
+    ary = imageStructToArray(imageRow)
+    # PIL expects RGB order, image schema is BGR
+    # => we need to flip the order unless there is only one channel
+    if imgType.nChannels != 1:
+        ary = _reverseChannels(ary)
+    if imgType.nChannels == 1:
+        return Image.fromarray(obj=ary,mode='L')
+    elif imgType.nChannels == 3:
+        return Image.fromarray(obj=ary,mode='RGB')
+    elif imgType.nChannels == 4:
+        return Image.fromarray(obj=ary,mode='RGBA')
     else:
-        raise ValueError("did not find a sparkMode for the given array with num_channels = %d " +
-                         "and dtype %s" % (num_channels, arr.dtype.string))
+        raise ValueError("don't know how to convert " + imgType.name + " to PIL")
 
+def PIL_to_imageStruct(img):
+    # PIL is RGB based, image schema expetcs BGR ordering => need to flip the channels
+    return _reverseChannels(np.asarray(img))
 
-def _resizeFunction(size):
-    """ Creates a resize function.
-    
-    :param size: tuple, size of new image: (height, width). 
-    :return: function: image => image, a function that converts an input image to an image with 
-    of `size`.
-    """
 
-    if len(size) != 2:
-        raise ValueError("New image size should have for [hight, width] but got {}".format(size))
-
-    def resizeImageAsRow(imgAsRow):
-        imgAsArray = imageStructToArray(imgAsRow)
-        imgType = imageType(imgAsRow)
-        imgAsPil = Image.fromarray(imgAsArray, imgType.pilMode)
-        imgAsPil = imgAsPil.resize(size[::-1])
-        imgAsArray = np.array(imgAsPil)
-        return imageArrayToStruct(imgAsArray, imgType.sparkMode)
+def _arrayToOcvMode(arr):
+    assert len(arr.shape) == 3, "Array should have 3 dimensions but has shape {}".format(arr.shape)
+    num_channels = arr.shape[2]
+    if arr.dtype == "uint8":
+        name = "CV_8UC%d" % num_channels
+    elif arr.dtype == "float32":
+        name = "CV_32FC%d" % num_channels
+    else:
+        raise ValueError("Unsupported type '%s'" % arr.dtype)
+    return imageTypeByName(name)
 
-    return resizeImageAsRow
 
+def _reverseChannels(ary):
```

Comment:
I disagree with this one. They do the same thing so it should be one method in my opinion. You can always add comment to the call site. Otherwise it's really easy to use the wrong name and really confuse people.